src/SnackbarProvider/SnackbarProvider.tsx

```diff
@@ -307,14 +307,44 @@ class SnackbarProvider extends Component<SnackbarProviderProps, State> {
         const snackbars = Object.keys(categ).map((origin) => {
             const snacks = categ[origin];
             const [nomineeSnack] = snacks;
+
+             const categorizedSnacks: { [category: string]: InternalSnack[] } = {
+                call: [],
+                notification: [],
+                notificationViewAll: [],
+                regular: [],
+            };
+
+            snacks.forEach((snack: InternalSnack) => {
+                const snackId = String(snack.id);
+                let category = 'regular';
+
+                if (snackId.startsWith('call-')) {
+                    category = 'call';
+                } else if (snackId.startsWith('notification-')) {
+                    category = 'notification';
+                } else if (snackId === 'view_all_notifications') {
+                    category = 'notificationViewAll';
+                }
+
+                categorizedSnacks[category].push(snack);
+            });
+
+            const combinedSnacks = [
+                ...categorizedSnacks['call'],
+                ...categorizedSnacks['notification'].reverse(),
+                ...categorizedSnacks['notificationViewAll'],
+                ...categorizedSnacks['regular'],
+            ];
+
```

Comment:
In order to maintain the independence of the library, it is better to be able to explicitly specify the display order from outside the library than to have a behavior that is invisible from outside the library.

Therefore, it would be a good idea to add the following option? 


```diff
diff --git a/src/SnackbarProvider/SnackbarProvider.tsx b/src/SnackbarProvider/SnackbarProvider.tsx
index c404716..e39e414 100644
--- a/src/SnackbarProvider/SnackbarProvider.tsx
+++ b/src/SnackbarProvider/SnackbarProvider.tsx
@@ -104,6 +104,7 @@ class SnackbarProvider extends Component<SnackbarProviderProps, State> {
             style: merger('style', true),
             SnackbarProps: merger('SnackbarProps', true),
             className: clsx(this.props.className, options.className),
+            displayOrder: options.displayOrder ?? 0,
         };
 
         if (snack.persist) {
@@ -306,7 +307,7 @@ class SnackbarProvider extends Component<SnackbarProviderProps, State> {
 
         const snackbars = Object.keys(categ).map((origin) => {
             const snacks = categ[origin];
-            const [nomineeSnack] = snacks;
+            const [nomineeSnack] = snacks.sort((a, b) => a.displayOrder - b.displayOrder);
             return (
                 <SnackbarContainer
                     key={origin}
diff --git a/src/types.ts b/src/types.ts
index 54dbe7d..c6e8b83 100644
--- a/src/types.ts
+++ b/src/types.ts
@@ -300,6 +300,12 @@ export interface OptionsObject<V extends VariantType = VariantType> extends Shar
      * @default false
      */
     persist?: boolean;
+
+    /**
+     * Order of a display. The order is switched only in the snackbar that is already displayed
+     * @default 0
+     */
+    displayOrder?: number;
 }
 
 /** Properties of the internal snack which should not be exposed to out
side world  */
@@ -318,7 +324,8 @@ type NeededByInternalSnack =
     | 'TransitionProps'
     | 'transitionDuration'
     | 'hideIconVariant'
-    | 'disableWindowBlurListener';
+    | 'disableWindowBlurListener'
+    | 'displayOrder';
 
 /**
  * Properties of a snackbar internal to notistack implementation. Not t
o be used by outside

```

Reply:
@satoren I have added your suggested changes, and updated the code to support the ability to show user notifications snackbars sorted in descending order.